openedx/core/djangoapps/schedules/management/commands/__init__.py

```diff
@@ -22,10 +22,17 @@ def add_arguments(self, parser):
             help='Send all emails to this address instead of the actual recipient'
         )
         parser.add_argument('site_domain_name')
+        parser.add_argument(
+            '--duration',
+            help='The duration for which weekly emails are sent (days).',
+        )
 
     def handle(self, *args, **options):
         self.log_debug('Args = %r', options)
 
+        if 'duration' in options.keys():
+            self.offsets = xrange(-7, -options['duration'], -7)
+
```

Comment:
Perhaps we can have a check for the type of data passed in `duration` option before using it in xrange? In case of wrong data type, we can have a clear error message.

Reply:
@pkulkark absolutely. I've added a type specification for 'int' in the parser method.